library/core/src/main/java/com/google/android/exoplayer2/audio/DefaultAudioSink.java

```diff
@@ -342,15 +365,20 @@ public void configure(@C.Encoding int inputEncoding, int inputChannelCount, int
     int channelCount = inputChannelCount;
     int sampleRate = inputSampleRate;
     isInputPcm = isEncodingPcm(inputEncoding);
+    shouldUpResPCMAudio = canConvertHiResPcmToFloat &&
+     (inputEncoding == C.ENCODING_PCM_24BIT || inputEncoding == C.ENCODING_PCM_32BIT);
     if (isInputPcm) {
-      pcmFrameSize = Util.getPcmFrameSize(inputEncoding, channelCount);
+      pcmFrameSize = Util.getPcmFrameSize(shouldUpResPCMAudio
+       ? C.ENCODING_PCM_FLOAT : inputEncoding, channelCount);
     }
     @C.Encoding int encoding = inputEncoding;
     boolean processingEnabled = isInputPcm && inputEncoding != C.ENCODING_PCM_FLOAT;
```

Comment:
For the 24-bit integer PCM -> float path we can't apply playback parameters, but processingEnabled gets set to true so `setPlaybackParameters` will try to apply the playback parameters. For now shall we just add another flag `canApplyPlaybackParams` which is `processingEnabled && !shouldUpResPCMAudio` and change the check in `setPlaybackParameters` to use that?